proposals/0001-unify-nat-and-natural.md

```diff
@@ -0,0 +1,71 @@
+---
+author: Rinat Stryungis
+date-accepted: ""
+ticket-url: "https://gitlab.haskell.org/ghc/ghc/-/issues/10776"
+implemented: "Yes"
+---
+
+This proposal is [discussed at this pull request](https://github.com/haskell-core/core-libraries-proposals/pull/0>).
+**After creating the pull request, edit this file again, update the number in
+the link, and delete this bold sentence.**
+
+# Proposal title
+
+The changes in the proposal remove the separate built-in kind `Nat` 
+in favor of promoted type `Natural`.
+
+
+## Motivation
+
+Existence of separated kind `Nat` and impossibility of promotion 
+of usual `Natural` data type (and data constructors with `Natual` fields)
+looks like an unnecessary and inconvenient complication. 
+It forces to create redundant types with `Nat` fields in their constructors (it makes these types uninhabited in terms)
+only for promotional purposes. This is more cumbersome and less safe than the promotion 
+of the ordinary data type. 
+
+## Proposed Change Specification
+
+1. This patch removes the separate kind `Nat` and repaces it by promoted type `Natural`. 
+2. Allows promoting of the data constructors with Natural numbers in their fields. 
+3. Module `GHC.TypeNats` now contains type synonym `type Nat = Natural` for the backward compatibility 
+
+## Examples
+
+The pretty simple example: 
+
+Before this change, we had to create a separate type for the promotion
+
+    data C = MkC Natural
+    data CP = MkCP Nat
+    type CC = MkCP 1
```

Comment:
Let's make it clear why `CP` is needed here by showing which terms don't typecheck.

Before the proposal:

```
data C = MkC Natural
c1 = MkC 1        -- works
type C1 = MkC 1   -- doesn't work

data P = MkP Nat
p1 = MkP 1        -- doesn't work
type P1 = MkP 1   -- works
```

After the proposal:

```
data C = MkC Natural
c1 = MkC 1       -- works
type C1 = MkC 1  -- works
```